nodes/src/nodes/tool_filesystem/IInstance.py

```
@@ -0,0 +1,328 @@
+        },
+        description=('List the immediate children of a directory in the account file store. Optional: "path" (defaults to the account root). Returns: {entries: [{name, type, size?, modified?}], count}.'),
+    )
+    def list_directory(self, args):
+        args = _require_dict(args) if args is not None else {}
+        self._check_ready()
+        if not self.IGlobal.allow_list:
+            raise ValueError('list access is not enabled for this filesystem tool')
+
+        path = args.get('path', '')
+        if not isinstance(path, str):
+            raise ValueError('path must be a string')
```

Ruff (0.15.10)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)